core/src/main/java/de/danoeh/antennapod/core/syndication/parsers/DurationParser.java

```diff
@@ -0,0 +1,37 @@
+package de.danoeh.antennapod.core.syndication.parsers;
+
+import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class DurationParser {
+    public static long inMillis(String durationStr) {
+        String[] parts = durationStr.trim().split(":");
+
+        if (parts.length == 1) {
+            return toMillis(parts[0]);
+        } else if (parts.length == 2) {
+            return toMillis("0", parts[0], parts[1]);
+        } else if (parts.length == 3) {
+            return toMillis(parts[0], parts[1], parts[2]);
+        } else {
+            throw new NumberFormatException();
+        }
+    }
+
+    static private long toMillis(String hours, String minutes, String seconds) {
+        return HOURS.toMillis(Long.parseLong(hours))
+                + MINUTES.toMillis(Long.parseLong(minutes))
+                + toMillis(seconds);
+    }
+
+    static private long toMillis(String seconds) {
```

Comment:
Please swap the order to `private static` like in the [google code style guide](https://google.github.io/styleguide/javaguide.html#s4.8.7-modifiers)